Editor/SettingsAnalysis/UniversalRenderPipelineAnalyzer.cs

```diff
@@ -0,0 +1,182 @@
+        );
+
+        public void Initialize(ProjectAuditorModule module)
+        {
+            module.RegisterDescriptor(k_URPAssetDescriptor);
+            module.RegisterDescriptor(k_HdrSettingDescriptor);
+            module.RegisterDescriptor(k_MsaaSampleCountSettingDescriptor);
+            module.RegisterDescriptor(k_CameraStopNanDescriptor);
+        }
+
+        public IEnumerable<ProjectIssue> Analyze(ProjectAuditorParams projectAuditorParams)
+        {
+#if UNITY_2019_3_OR_NEWER && PACKAGE_URP
+            var renderPipeline = GraphicsSettings.currentRenderPipeline;
+            if (renderPipeline == null || renderPipeline is not UniversalRenderPipelineAsset)
+            {
+                yield return ProjectIssue.Create(IssueCategory.ProjectSetting, k_URPAssetDescriptor)
+                    .WithLocation("Project/Graphics");
+            }
+            else
+            {
+#if UNITY_ANDROID || UNITY_IOS || UNITY_SWITCH
+                foreach (var analyzeSrpAsset in RenderPipelineUtils.AnalyzeAssets(Analyze)) yield return analyzeSrpAsset;
+
+                var allCameraData = RenderPipelineUtils
+                    .GetAllComponents<UniversalAdditionalCameraData>();
+                foreach (var cameraData in allCameraData)
+                {
+                    if (cameraData.stopNaN)
+                        yield return ProjectIssue.Create(IssueCategory.ProjectSetting,
+                            k_CameraStopNanDescriptor);
+                }
+#endif
+            }
+#else
+            yield break;
+#endif
+        }
+
+        private static void FixHdrSetting(ProjectIssue issue)
+        {
+#if UNITY_2019_3_OR_NEWER
+            RenderPipelineUtils.FixAssetSetting(issue, p => SetHdrSetting(p, false));
+#endif
+        }
+
+        private static void FixMsaaSampleCountSetting(ProjectIssue issue)
+        {
+#if UNITY_2019_3_OR_NEWER
+            RenderPipelineUtils.FixAssetSetting(issue, p => SetMsaaSampleCountSetting(p, 2));
+#endif
+        }
+
+#if UNITY_2019_3_OR_NEWER
+        private IEnumerable<ProjectIssue> Analyze(RenderPipelineAsset renderPipeline, int qualityLevel)
+        {
+#if PACKAGE_URP
+            bool? supportsHDR = GetHdrSetting(renderPipeline);
+            if (supportsHDR != null && supportsHDR.Value)
+            {
+                yield return RenderPipelineUtils.CreateAssetSettingIssue(qualityLevel, renderPipeline.name,
+                    k_HdrSettingDescriptor);
+            }
+
+            int? msaaSampleCount = GetMsaaSampleCountSetting(renderPipeline);
+            if (msaaSampleCount != null && msaaSampleCount >= 4)
+            {
+                yield return RenderPipelineUtils.CreateAssetSettingIssue(qualityLevel, renderPipeline.name,
+                    k_MsaaSampleCountSettingDescriptor);
+            }
+#else
+            yield break;
+#endif
+        }
+
+        internal static bool? GetHdrSetting(RenderPipelineAsset renderPipeline)
```

Comment:
Is there a particular need for this bool to be nullable? Seems to me that the code paths that return null could just as easily return false, and it would make the code easier to read.

Reply:
It was done to make clear if we actually get value, or render pipeline does not exist. But I guess in the end we don't need this info. So done.